feat: improve naming of downloaded traces

TitouanLarnicol opened the pull request:
Issue #3045 Renaming of the downloaded files: 

- For routes :  document ID + name of the summit + the name of the route
- For outing : document ID + name of the outing + date realized

I implemented the alternative described by [anakreenskyrider](https://github.com/anakreenskyrider) which remove the ID if the name if longer than 100 characters ( decided completely arbitrarily, might be too long ? )

brunobesson reviewed (commented):
All good for me, thanks @TitouanLarnicol!
I wonder if maybe we should handle spaces and special characters and replace them with e.g. underscores?
I assume it is platform and browser dependent, but I get e.g. `1483583_Grande Moucherolle _ Traversée des Moucherolles_2022-11-13.gpx`.
I feel something like `1483583_Grande_Moucherolle_Traversée_des_Moucherolles_2022-11-13.gpx` would be more appropriate? What do you think?

TitouanLarnicol commented:
Good idea @brunobesson , i will add that to the PR.

asaunier commented on src/views/document/utils/boxes/MapBox.vue:
```diff
@@ -142,13 +142,22 @@ export default {
     },
+
+    getDownloadedFileName(name, extension) {
+      const prefixRouteId = name.length < 100 ? this.document.document_id + '_' : '';
```
That would be more consistent to always prefix with the route id but substring the whole name at the first 100 characters if really to long?

asaunier replied:
`prefixDocumentId` would be more relevant as the document may be an outing :P

TitouanLarnicol replied:
> That would be more consistent to always prefix with the route id but substring the whole name at the first 100 characters if really to long?

it would be better indeed

TitouanLarnicol commented:
I let you check the changes. 
Added a little regex because there is always a ":" in the name of a route. I converted illegal arguments into spaces and replace every space with '_' to avoid names like : "Mont_Rachais___face_Est" (Browsers convert them into '_').
I can also remove the regex if you think it's better to let chrome adding those '_'.

brunobesson reviewed (approved):
Looks good to me, thanks!
